### PR TITLE
feature: isolation among all netavark bridge 

### PR DESCRIPTION
**Pull request by yassi-github**
Currently, it appears that isolation is only enabled when both bridges are in isolate mode.  
This means that communication with networks that do not have isolate enabled is still possible.

What about using `isolate=strict` to deny communication with networks that do not have isolate enabled?  
Specifically, add a new chain of iptables and add a DROP rule to the chain for networks that do not have isolate enabled.

Here is a table showing the connectivity between networks before and after this change:

before:

```
|send\recv  |nonisol1   |nonisol1   |isol1      |isol2      |
|-----------|-----------|-----------|-----------|-----------|
|nonisol1   |     o     |     o     |     o     |     o     |
|nonisol2   |     o     |     o     |     o     |     o     |
|isol1      |     o     |     o     |     o     |     x     |
|isol2      |     o     |     o     |     x     |     o     |
```

after:

```
|send\recv  |nonisol1   |nonisol1   |isol1      |isol2      |strictisol1|strictisol2|
|-----------|-----------|-----------|-----------|-----------|-----------|-----------|
|nonisol1   |     o     |     o     |     o     |     o     |     x     |     x     |
|nonisol2   |     o     |     o     |     o     |     o     |     x     |     x     |
|isol1      |     o     |     o     |     o     |     x     |     x     |     x     |
|isol2      |     o     |     o     |     x     |     o     |     x     |     x     |
|strictisol1|     x     |     x     |     x     |     x     |     o     |     x     |
|strictisol2|     x     |     x     |     x     |     x     |     x     |     o     |
```

### CODE REVIEW

**Comment by yassi-github** [comment]
@Luap99 @flouthoc 
Is there anything else left to do for the review?
I'm new to OSS so sorry if I'm doing something wrong.

**Comment by Luap99** [comment]
@yassi-github No everything is good, I will try to take a look this week. I am a bit busy right now.

**Comment by yassi-github** [comment]
@Luap99 OK, Thank you for your time. :smiley:

**Comment by Luap99** [comment]
@mheon PTAL, can this be done with firewalld? I would like to avoid adding firewall driver specific options that we cannot support in other backends. Once we enable other backends we have to implement there as well. The problem is end users have no idea about what driver will be used and I really do not want critical security options only work with one implementation.

File: test/100-bridge-iptables.bats
Comment (Luap99) [inline test/100-bridge-iptables.bats:683]: Please add comments here what checks what, i.e from network x to y. If one thing here fails it is a nightmare to debug.

File: test/100-bridge-iptables.bats
Comment (Luap99) [inline test/100-bridge-iptables.bats:702]: same here

**Review by Luap99 — COMMENTED** [review]
Also could you squash the commits please.

**Comment by Luap99** [comment]
@mheon What is the state of firewalld implementation then? I guess this is already a problem with the current isolate=true/false behavior? I think this needs to be supported by firewalld if we ever want to enable firewalld. But given how long we have been pushing this I have doubts that this is going to happen soon.  

In any case I think this PR is good and solves a real problem and the firewalld future is something for us maintainers to figure out and should not block this PR from merging.

**Review by Luap99 — APPROVED** [review]
LGTM, thanks
/lgtm

In order to create networks with `podman network create --opt isolate=strict` we need to patch c/common to accept that new value.
Do you want to make a PR there to enable it? If not I can do it. 

ref: https://github.com/containers/common/blob/20def0054c6e1841e1b33fe4ecc45c0ab3c29e6f/libnetwork/netavark/config.go#L189-L195

**Comment by yassi-github** [comment]
> LGTM, thanks /lgtm

Glad I could contribute!

> In order to create networks with `podman network create --opt isolate=strict` we need to patch c/common to accept that new value. Do you want to make a PR there to enable it? If not I can do it.
> 
> ref: https://github.com/containers/common/blob/20def0054c6e1841e1b33fe4ecc45c0ab3c29e6f/libnetwork/netavark/config.go#L189-L195

I'd like to try it.

**Pull request merged**
